client/utils.py

```diff
@@ -0,0 +1,44 @@
+# The MIT License (MIT)
+#
+# Copyright (c) 2021 NVIDIA CORPORATION
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy of
+# this software and associated documentation files (the "Software"), to deal in
+# the Software without restriction, including without limitation the rights to
+# use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+# the Software, and to permit persons to whom the Software is furnished to do so,
+# subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+# FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+# COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+# IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+# CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+import numpy as np
+
+def batcher(dataset, batch_size, n_iterations=-1):
```

Comment:
You could adjust this to support random batch size (like in https://github.com/triton-inference-server/dali_backend/blob/2d5fe4cf7532efb11e81fdfa6272585c3009dd79/qa/identity/identity_client.py#L60 and other qa clients) e.g. by adding max_batch_size and min_batch_size and then use this batcher in those qa clients. 
Maybe not in the scope of the PR but it's about refactoring and doing it would remove some code repetition in the qa clients.

Reply:
Boosted up the `batcher`. Now it takes callable, list or scalar.

We settled up offline, that I won't replace the `batcher` in other scripts, since they have open PRs right now. But certainly that would be the next step